### PR TITLE
Create dialogs with custom button rows

### LINKED ISSUE #2782: dialog.NewProgressInfinite is deprecated, but dialog.NewCustom isn't equivalent
### Is your feature request related to a problem? Please describe:

`dialog.NewProgressInfinite()` uses a trick to get rid of dismiss button, but `dialog.NewCustom()` doens't give such possibility.

### Is it possible to construct a solution with the existing API?

Still use `dialog.NewProgressInfinite()`.

### Describe the solution you'd like to see:

The simplest way is to not create dismiss button if `dismiss string` parameter of `dialog.NewCustom()` is empty.
Or maybe version like `dialog.NewCustomWithButtons()` or event expose `setButtons()` method.

### PR DESCRIPTION
**Pull request by Jacalz**
### Description:

This implements the changes needed to be able to create buttons without buttons as well as to use custom button rows.
The deprecation comments for the progressbar dialogs have been updated to mention this new API. 
To make things more consistent, all dialogs now use grids for the button row.

Fixes  #2782, #2127

### Checklist:

- [x] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

#### Where applicable:

- [x] Public APIs match existing style and have Since: line.

### CODE REVIEW

**Review by andydotxyz — CHANGES_REQUESTED** [review]
Overall this looks great, just a couple of small things to tweak.

**Comment by Jacalz** [comment]
I had to force-push to get the conflict resolved. Everything should be good now :)

File: dialog/custom.go [hunk 1]
```diff
@@ -3,24 +3,28 @@ package dialog
...
 	"fyne.io/fyne/v2/widget"
 )
 
+// Custom implements a custom dialog.
+//
+// Since 2.4
+type Custom struct {
+	*dialog
+}
+
```
[the remark this answers is not in the document]
Comment (andydotxyz) [reply dialog/custom.go:18]: Sorry I didn't mean a rename - (`dialog.Custom` was clear enough) it's the missing `var _ Dialog = (*Custom)(nil)`

**Review by andydotxyz — APPROVED** [review]
I wasn't sure about the `Custom` -> `CustomDialog` change, but I see it matches the other types in the package so probably good.

**Comment by Jacalz** [comment]
The name change was so that we theoretically could add a `Custom` type for the `v2` dialogs in the future (as discussed on the call a few weeks back). On top of that, it is as you say, more in line with the other types :)

File: dialog/custom.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jacalz) [reply dialog/custom.go:18]: Indeed. I know. I added the check as well. The rename was for another purpose as per my other comment :)

**Comment by Jacalz** [comment]
In all honesty, I am not entirely sure how the new dialogs will pan out. I can can revert the name change if you'd like?

File: dialog/custom.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (andydotxyz) [reply dialog/custom.go:18]: I missed the other comment somehow, thanks for getting this all sorted

**Comment by andydotxyz** [comment]
> In all honesty, I am not entirely sure how the new dialogs will pan out. I can can revert the name change if you'd like?

I think it is consistent with released API names in the package, in which case it should probably stay?

**Review by andydotxyz — APPROVED** [review]

**Comment by Jacalz** [comment]
Yeah, I agree that consistency wins. I think it makes sense to keep it as it is now and do any aliasing or naming improvements as part of `v2` once it gets traction :)

**Pull request merged**